Sources/NIOPosix/NonBlockingFileIO.swift

```
@@ -384,28 +384,63 @@ public struct NonBlockingFileIO: Sendable {
     ) throws -> ByteBuffer {
         var bytesRead = 0
         var buf = allocator.buffer(capacity: byteCount)
+
+        #if os(Windows)
+        // Windows does not offer pread. Because of this we have to
+        // 1. get the current offset in the fd
+        // 2. change the offset to the value we want
+        // 3. read
+        // 4. reset the offset to the og value
+        if let fromOffset {
+            let ogOffset = try fileHandle.withUnsafeFileDescriptor { descriptor -> off_t in
+                let ogOffset = try Posix.lseek(
+                    descriptor: descriptor,
+                    offset: 0,
+                    whence: SEEK_SET
+                )
+                try Posix.lseek(
+                    descriptor: descriptor,
+                    offset: off_t(fromOffset),
+                    whence: SEEK_SET
+                )
+                return ogOffset
+            }
+            defer {
+                // Swallow any error here?
+                _ = try? fileHandle.withUnsafeFileDescriptor { descriptor in
+                    try Posix.lseek(
+                        descriptor: descriptor,
+                        offset: ogOffset,
+                        whence: SEEK_SET
+                    )
+                }
+            }
```

defer swallows any errors here. so defer might not work for us. @Lukasa wdyt?